packages/rich-text/src/component/use-anchor.js

```diff
@@ -162,7 +169,19 @@ export function useAnchor( { editableContentElement, settings = {} } ) {
 				getAnchor( editableContentElement, tagName, className )
 			);
 		}
-	}, [ editableContentElement, tagName, className, isActive, wasActive ] );
+	}, [
+		editableContentElement,
+		tagName,
+		className,
+		isActive,
+		wasActive,
+		// Active attributes is defaulted to a stable object to avoid re-rendering,
+		// but in specific circumstances it can be used to provide additional information
+		// about the currently active format to disambiguate the format from other instances of the same format
+		// within the same editable content element. This is useful for `core/link` when you
+		// want to disambiguate between different links when clicking between them.
```

Comment:
Reading this comment, and looking at the history of this file, I see that in 6.4 we had event listeners to listen to "selection changes". I'm almost certain that these event listeners were here to address this particular issue. (update the anchor when the selection changes).

Reply:
Discussion continues in [this comment](https://github.com/WordPress/gutenberg/pull/59635#discussion_r1517687396)

Reply:
This PR explains the rationale and bug fixes it addresses: https://github.com/WordPress/gutenberg/pull/58900

tl;dr: The selection listeners were necessary for the previous link popover behavior of using the arrows to move a caret into the text to show the link ui popover. This UX was changed based on design and a11y feedback, so they were no longer necessary. Also, the previous useAnchor implementation did not account for repositioning a popover when the selection changed elements underneath of it (for example, when creating a link, the selected text goes from no extra markup to having an <a> wrapped on it, which caused the popover to lose its anchoring).

Reply:
> The selection listeners were necessary for the previous link popover behavior of using the arrows to move a caret into the text to show the link ui popover.

So I _think_ that may have been there also to account for clicking between links. So when you click from one link to the next the "selection" event fires and thus the anchor gets recompute.